src/ChannelWrapper.js

```diff
@@ -348,29 +362,27 @@ export default class ChannelWrapper extends EventEmitter {
                 }
             })();
 
+            // Send some more!
+            this._publishQueuedMessages(workerNumber);
+
             return sendPromise;
         })
         .then(
             result => {
-                this._messages.shift();
+                this._unconfirmedMessages.shift();
```

Comment:
This assumes messages will be confirmed in the same order they were added to _unconfirmedMessages, but that sounds reasonable.